src_compressible/flux.cpp

```diff
@@ -1239,6 +2010,16 @@ void calculateFlux(const MultiFab& cons_in, const MultiFab& prim_in,
                 for (int l=0; l<nspecies+6; ++l) {
                     primitive[l] = wgt1*(prim(i,j,k,l)+prim(i-1,j,k,l)) - wgt2*(prim(i-2,j,k,l)+prim(i+1,j,k,l));
                 }
+                if ((i == 0) and (bc_mass_lo[0] != 3) and (bc_mass_lo[0] != -1)) { // ignore for reservoirs and periodic BC
```

Comment:
This is a very subtle point, and applies both here and in the conservative variable interpolation to faces.  If i==1 then the 4 point stencil isn't quite right since the i=-1 point isn't space where the stencil thinks it should be.  @jbbel should we just use 2 point in that case, or derive the 3rd-order stencil involving those 4 point?

Reply:
fixed this to include modified stencil one face in from the boundary